### PR TITLE
Fix flaky TestStalePodDisruption

### LINKED ISSUE #112594: Failure cluster [f6cf3d64...]: TestStalePodDisruption
##### Error text:
```
=== RUN   TestStalePodDisruption/stale_pod_disruption
W0903 05:46:12.224218   76126 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0903 05:46:12.226482   76126 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0903 05:46:12.227136   76126 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0903 05:46:12.227260   76126 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0903 05:46:12.227326   76126 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
W0903 05:46:12.227432   76126 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
I0903 05:46:12.329811   76126 disruption.go:421] Sending events to api server.
I0903 05:46:12.334312   76126 disruption.go:432] Starting disruption controller
I0903 05:46:12.334420   76126 shared_informer.go:255] Waiting for caches to sync for disruption
I0903 05:46:12.334571   76126 shared_informer.go:262] Caches are synced for disruption
    disruption_test.go:1529: Obtained pod conditions (-want,+got):
          []v1.PodCondition{
          	{
          		Type:          "DisruptionTarget",
        - 		Status:        "False",
        + 		Status:        "True",
          		LastProbeTime: {},
          		... // 1 ignored and 2 identical fields
          	},
          }
I0903 05:46:12.437389   76126 disruption.g
```
#### Recent failures:
[2022/9/8 04:20:19 ci-kubernetes-unit](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-unit/1567608212939608064)
[2022/9/3 13:25:13 ci-kubernetes-generate-make-test-count10](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-generate-make-test-count10/1565933915070271488)

/kind failing-test

/kind flake

/sig testing apps

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind bug
/kind flake

#### What this PR does / why we need it:

We cannot trust the length of the work queue to determine if the pod has already being processed. The pod might not have entered the queue or it might still might be being processed.

The only option is to poll the Pod status itself.

#### Which issue(s) this PR fixes:

Fixes #112594

#### Special notes for your reviewer:

We need to cherry-pick to 1.25.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/disruption/disruption_test.go [hunk 1]
```diff
@@ -1511,22 +1511,20 @@ func TestStalePodDisruption(t *testing.T) {
 			if _, err := dc.coreClient.CoreV1().Pods(tc.pod.Namespace).Create(ctx, tc.pod, metav1.CreateOptions{}); err != nil {
 				t.Fatalf("Failed to create pod: %v", err)
 			}
+			dc.clock.Sleep(tc.timePassed)
 			if err := dc.informerFactory.Core().V1().Pods().Informer().GetIndexer().Add(tc.pod); err != nil {
 				t.Fatalf("Failed adding pod to indexer: %v", err)
 			}
-			dc.clock.Sleep(tc.timePassed)
```
Comment (aojea) [inline pkg/controller/disruption/disruption_test.go:1517]: why the sleep order change?

Comment (alculquicondor) [reply pkg/controller/disruption/disruption_test.go:1517]: to ensure the time passed before the object is processed by the worker

**Review by soltysh — APPROVED** [review]
/approve

**Pull request merged**